### DISCUSSION TITLE
Dark mode - UI and "white mode"

### DISCUSSION THREAD

**Discussion by katepolak** [voxeltycoon/issues#395] [UI&UX]
**Is your feature request related to a problem? Please describe.**
I am one of those guys who has dark mode enabled on everything, so my eyes burn all the time playing VT. But don' worry, VT is so good it's worth it 😉 

**Describe the solution you'd like**
I would love to see a dark mode of the UI, but more importantly a dark mode of the "white mode", from a quick look at the code that one shouldn't even be too hard.

**Describe alternatives you've considered**
No alternatives available I would say.

**Additional context**
This one is pretty self explanatory I would say.

**Comment by Hollikill** [comment]
maybe add a feature that lest the player pick from a range of colors for the UI 
(similar to color picker in asset editor or company color picker)

**Comment by katepolak** [comment]
That would be quite hard, if not impossible, to make work. Not from the technical side of things but to make all the different color combinations work.

**Comment by victorleblais** [comment]
With some math trickery, you can make almost any color work in order to have a readable UI.

You could let the user pick a color, then automatically generate darker or lighter shades for the different sections of the UI.

You can mathematically determine if a color is "dark or light" in order to change the color of the text (e.g. the color selector for Github's label does that).

Sure, some colors would be "less nice" than others (e.g. if you select pitch black, then you can't get a darker shade), but it can work nonetheless.

As to determine if it's a good idea. I don't know. I'm all for customization where possible.

At the very least, exposing all the UI colors in a JSON file could be a nice alternative to allow more "experienced" users to mess about with the color scheme of the entire UI.
